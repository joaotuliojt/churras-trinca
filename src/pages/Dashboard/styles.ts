import { styled } from '@styles/stitches'

export const Container = styled('main', {
  padding: '0 4.8rem',
  marginTop: '6rem'
})

export const PageDescription = styled('div', {
  h1: {
    fontWeight: 700,
    fontSize: '4rem',
    marginBottom: '.8rem'
  },
  p: {
    fontWeight: 500,
    fontSize: '1.6rem',
    maxWidth: '53.2rem'
  }
})

export const GridContainer = styled('section', {
  display: 'grid',
  gridTemplateColumns: 'repeat(5, 1fr)',
  marginTop: '3rem',
  gap: '1.6rem',
  rowGap: '3rem',
  '@media(max-width: 1440px)': {
    gridTemplateColumns: 'repeat(4, 1fr)'
  },
  '@media(max-width: 1080px)': {
    gridTemplateColumns: 'repeat(3, 1fr)'
  },
  '@media(max-width: 820px)': {
    gridTemplateColumns: 'repeat(2, 1fr)'
  },
  '@media(max-width: 580px)': {
    gridTemplateColumns: '1fr'
  }
})

export const AddChurrasCard = styled('div', {
  display: 'grid',
  placeItems: 'center',
  backgroundColor: '$yellow',
  height: '20rem',
  borderRadius: '.6rem',
  cursor: 'pointer',
  boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
  div: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    gap: '1rem'
  },
  p: {
    fontSize: '1.8rem',
    fontWeight: 500
  }
})
