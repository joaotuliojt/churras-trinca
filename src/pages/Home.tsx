import { Main } from '../Components/Main'

export function Home() {
  return <Main />
}
