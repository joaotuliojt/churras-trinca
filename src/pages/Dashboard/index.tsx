import { BarbecueCard } from '@/components/BarbecueCard'
import Header from '@/components/Header'
import { api } from '@/services/api'
import { useQuery } from '@tanstack/react-query'
import * as S from './styles'

interface IParticipant {
  name: string
  value: number
  withDrink: boolean
}

export interface IBarbecue {
  id: string
  date: Date
  description: string
  comments: string
  total: number
  values: {
    withDrinks: number
    withoutDrinks: number
  }
  participants: IParticipant[]
  totalParticipants: number
}

export function Dashboard() {
  const { data: barbecues } = useQuery<IBarbecue[]>({
    queryKey: ['barbecues'],
    queryFn: async () => {
      const { data } = await api.get<IBarbecue[]>(
        '/barbecues?_sort=date&_order=desc'
      )
      return data
    }
  })

  return (
    <>
      <Header />
      <S.Container>
        <S.PageDescription>
          <h1>Todos os churras marcados</h1>
          <p>
            Clique no primeiro card para adicionar um novo churras ou clique nos
            demais para visualizar o evento
          </p>
        </S.PageDescription>
        <S.GridContainer>
          <S.AddChurrasCard>
            <div>
              <img src="/assets/bbq_icon.svg" alt="" />
              <p>Adicionar Churras</p>
            </div>
          </S.AddChurrasCard>
          {barbecues?.map((barbecue) => (
            <BarbecueCard key={barbecue.id} {...barbecue} />
          ))}
        </S.GridContainer>
      </S.Container>
    </>
  )
}
