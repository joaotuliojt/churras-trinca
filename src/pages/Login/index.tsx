import { Button } from '../../Components/Button'
import { Input } from '../../Components/Input'
import * as S from './styles'
import * as yup from 'yup'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import { redirect } from 'react-router-dom'

interface LoginFormData {
  email: string
  password: string
}

const schema = yup.object({
  email: yup.string().email('E-mail inválido').required('Campo obrigatório'),
  password: yup.string().required('Campo obrigatório')
})

export function Login() {
  const { register, handleSubmit, formState } = useForm<LoginFormData>({
    resolver: yupResolver(schema)
  })
  const { errors, isSubmitting } = formState

  const onSubmitLogin = handleSubmit(async ({ email, password }) => {
    await new Promise((resolve) => setTimeout(resolve, 2000))
    console.log(email, password)
  })

  return (
    <S.Container>
      <S.LeftSide>
        <S.Logo src="assets/trinca.svg" />
      </S.LeftSide>
      <S.RightSide>
        <S.FormContainer>
          <S.HeadingWrap>
            <h2>Agenda de Churras</h2>
            <p>Informe seu login e senha para agendar seu incrível churrasco</p>
          </S.HeadingWrap>
          <S.LoginForm onSubmit={onSubmitLogin}>
            <S.InputWraps>
              <Input
                id="email"
                placeholder="E-mail"
                label="Login"
                error={errors.email?.message}
                {...register('email')}
              />
              <Input
                id="password"
                type="password"
                placeholder="Senha"
                label="Senha"
                error={errors.password?.message}
                {...register('password')}
              />
            </S.InputWraps>
            <Button type="submit" loading={isSubmitting}>
              Entrar
            </Button>
          </S.LoginForm>
        </S.FormContainer>
      </S.RightSide>
    </S.Container>
  )
}
