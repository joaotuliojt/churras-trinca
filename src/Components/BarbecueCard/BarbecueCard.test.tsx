import { IBarbecue } from '@/pages/Dashboard'
import { render, fireEvent } from '@testing-library/react'

import { BarbecueCard } from '.'

const barbecue: IBarbecue = {
  id: '1',
  date: new Date('2023-01-30T04:00:12.423Z'),
  description: 'Aniversário da Sara',
  comments: 'É uma festa fantasia',
  total: 120.37,
  values: {
    withDrinks: 30,
    withoutDrinks: 20
  },
  participants: [
    {
      name: 'André',
      value: 20,
      withDrink: false
    },
    {
      name: 'Sara',
      value: 30,
      withDrink: true
    },
    {
      name: 'Bruno',
      value: 30,
      withDrink: true
    },
    {
      name: 'Roberto',
      value: 20,
      withDrink: false
    },
    {
      name: 'Carol',
      value: 20,
      withDrink: false
    }
  ],
  totalParticipants: 5
}

describe('<BarbecueCard />', () => {
  it('should translate on hover', () => {
    const { container } = render(<BarbecueCard {...barbecue} />)

    fireEvent.mouseEnter(container)
    //expect(container).toHaveStyle('transform: translateY(-1rem);')
    expect(container.firstChild).toMatchSnapshot()
  })
})
