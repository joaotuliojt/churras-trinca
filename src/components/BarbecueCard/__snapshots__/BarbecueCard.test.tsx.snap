// Vitest Snapshot v1

exports[`<BarbecueCard /> > should translate on hover 1`] = `
<div
  class="c-kwCyfN"
>
  <div
    class="c-fixGjY"
  >
    <div
      class="c-bxJlOZ"
    >
      <svg
        fill="currentColor"
        height="24"
        stroke-width="1"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <rect
          fill="none"
          height="176"
          rx="8"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          width="176"
          x="40"
          y="40"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="176"
          x2="176"
          y1="24"
          y2="56"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="80"
          x2="80"
          y1="24"
          y2="56"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="40"
          x2="216"
          y1="88"
          y2="88"
        />
      </svg>
      <p>
        <time
          datetime="1675051212423"
        >
          30/01
        </time>
      </p>
    </div>
    <p
      class="c-iDHzzj"
    >
      Anivers√°rio da Sara
    </p>
  </div>
  <div
    class="c-VxzFO"
  >
    <div
      class="c-egGYGd"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <circle
          cx="88"
          cy="108"
          fill="none"
          r="52"
          stroke="currentColor"
          stroke-miterlimit="10"
          stroke-width="16"
        />
        <path
          d="M155.4,57.9A54.5,54.5,0,0,1,169.5,56a52,52,0,0,1,0,104"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <path
          d="M16,197.4a88,88,0,0,1,144,0"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <path
          d="M169.5,160a87.9,87.9,0,0,1,72,37.4"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
      5
    </div>
    <div
      class="c-gForkO"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="128"
          x2="128"
          y1="72"
          y2="88"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="128"
          x2="128"
          y1="168"
          y2="184"
        />
        <circle
          cx="128"
          cy="128"
          fill="none"
          r="96"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <path
          d="M104,168h36a20,20,0,0,0,0-40H116a20,20,0,0,1,0-40h36"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
      <p
        class="c-cJbzxd"
      >
        R$ 120,37
      </p>
    </div>
  </div>
</div>
`
